hbase-client/src/main/java/org/apache/hadoop/hbase/client/MasterRegistry.java

```diff
@@ -79,36 +51,27 @@
  * TODO: Handle changes to the configuration dynamically without having to restart the client.
  */
 @InterfaceAudience.Private
-public class MasterRegistry implements ConnectionRegistry {
+public class MasterRegistry extends AbstractRpcBasedConnectionRegistry {
 
   /** Configuration key that controls the fan out of requests **/
   public static final String MASTER_REGISTRY_HEDGED_REQS_FANOUT_KEY =
     "hbase.client.master_registry.hedged.fanout";
 
-  /** Default value for the fan out of hedged requests. **/
-  public static final int MASTER_REGISTRY_HEDGED_REQS_FANOUT_DEFAULT = 2;
+  public static final String MASTER_REGISTRY_PERIODIC_REFRESH_INTERVAL_SECS =
+    "hbase.client.master_registry.refresh_interval_secs";
 
-  private static final String MASTER_ADDRS_CONF_SEPARATOR = ",";
-
-  private final int hedgedReadFanOut;
-
-  // Configured list of masters to probe the meta information from.
-  private volatile ImmutableMap<ServerName, ClientMetaService.Interface> masterAddr2Stub;
-
-  // RPC client used to talk to the masters.
-  private final RpcClient rpcClient;
-  private final RpcControllerFactory rpcControllerFactory;
-  private final int rpcTimeoutMs;
+  public static final String MASTER_REGISTRY_MIN_SECS_BETWEEN_REFRESHES =
+    "hbase.client.master_registry.min_secs_between_refreshes";
 
-  protected final MasterAddressRefresher masterAddressRefresher;
+  private static final String MASTER_ADDRS_CONF_SEPARATOR = ",";
 
   /**
    * Parses the list of master addresses from the provided configuration. Supported format is comma
    * separated host[:port] values. If no port number if specified, default master port is assumed.
    * @param conf Configuration to parse from.
    */
-  private static Set<ServerName> parseMasterAddrs(Configuration conf) throws UnknownHostException {
-    Set<ServerName> masterAddrs = new HashSet<>();
+  public static List<ServerName> parseMasterAddrs(Configuration conf) throws UnknownHostException {
```

Comment:
nit: why set to list? Set was used to avoid duplicates fwiw.

Reply:
Reverted to Set.